src/main/java/net/openhft/chronicle/wire/WireMarshaller.java

```diff
@@ -488,14 +488,26 @@ public T defaultValue() {
     public void reset(T o) {
         try {
             for (FieldAccess field : fields) {
-                field.copy(defaultValue, o);
+                if (canBeReset(field)) {
+                    Object value = field.field.get(o);
+                    if (value != null)
+                        Wires.reset(value);
+                } else
+                    field.copy(defaultValue, o);
             }
         } catch (IllegalAccessException e) {
             // should never happen as the types should match.
             throw new AssertionError(e);
         }
     }
 
+    private static boolean canBeReset(FieldAccess field) {
```

Comment:
This feels like it should be a method in FieldAccess so it can be optimised.
`field.resetWith(defaultValue, o)`